nixos/modules/hardware/video/nvidia.nix

```
@@ -343,14 +343,20 @@ in
             ];
 
             # Don't add `nvidia-uvm` to `kernelModules`, because we want
-            # `nvidia-uvm` be loaded only after `udev` rules for `nvidia` kernel
-            # module are applied.
+            # `nvidia-uvm` be loaded only after the GPU device is available, i.e. after `udev` rules
+            # for `nvidia` kernel module are applied.
+            # This matters on Azure GPU instances: https://github.com/NixOS/nixpkgs/pull/267335
             #
             # Instead, we use `softdep` to lazily load `nvidia-uvm` kernel module
             # after `nvidia` kernel module is loaded and `udev` rules are applied.
             extraModprobeConfig = ''
               softdep nvidia post: nvidia-uvm
             '';
+
+            # Exception is the open-source kernel module failing to load nvidia-uvm using softdep,
+            # for unknown reasons.
+            # This breaks CUDA: https://github.com/NixOS/nixpkgs/issues/334180
+            kernelModules = lib.optionals cfg.open [ "nvidia-uvm" ];
```

I suggest `nvidia-uvm` -> `nvidia_uvm` for naming consistency with the other modules, else I'd be:

```
  "nvidia-uvm"
  "nvidia"
  "nvidia_modeset"
  "nvidia_drm"
```

```suggestion
            kernelModules = lib.optionals useOpenModules [ "nvidia_uvm" ];
```